activerecord/lib/active_record/relation/batches/batch_enumerator.rb

```diff
@@ -88,13 +88,15 @@ def touch_all(...)
         end
       end
 
-      # Destroys records in batches.
+      # Destroys records in batches. Returns the total number of rows affected.
       #
       #   Person.where("age < 10").in_batches.destroy_all
       #
       # See Relation#destroy_all for details of how each batch is destroyed.
       def destroy_all
-        each(&:destroy_all)
+        sum do |relation|
+          relation.destroy_all.size
```

Comment:
If what we want is the total number of rows "affected", we should count `destroyed` records, since `destroy_all` returns matched records whether a record is destroyed or not. ref https://github.com/rails/rails/pull/37782#issuecomment-571157660

```suggestion
          relation.destroy_all.count(&:destroyed?)
```

Reply:
You are right, thanks! Also updated the test case to account for this.